lib/matplotlib/artist.py

```diff
@@ -320,22 +320,34 @@ def get_window_extent(self, renderer):
         """
         return Bbox([[0, 0], [0, 0]])
 
-    def _get_clipping_extent_bbox(self):
-        """
-        Return a bbox with the extents of the intersection of the clip_path
-        and clip_box for this artist, or None if both of these are
-        None, or ``get_clip_on`` is False.
-        """
-        bbox = None
-        if self.get_clip_on():
-            clip_box = self.get_clip_box()
-            if clip_box is not None:
-                bbox = clip_box
-            clip_path = self.get_clip_path()
-            if clip_path is not None and bbox is not None:
-                clip_path = clip_path.get_fully_transformed_path()
-                bbox = Bbox.intersection(bbox, clip_path.get_extents())
-        return bbox
+    def _is_axes_clipped(self):
+        """
+        Return a boolean indicating whether the artist is clipped by the
+        axes bounding box or patch. True if ``get_clip_on`` is True, one of
+        `clip_box` or `clip_path` is set, ``clip_box.extents`` is equivalent
+        to ``axes.bbox.extents`` (if set), and ``clip_path._patch`` is
+        equivalent to ``axes.patch`` (if set).
+        """
+        # Note that ``clip_path.get_fully_transformed_path().get_extents()``
+        # cannot be directly compared to ``axes.bbox.extents`` because the
+        # extents may be undefined (i.e. positive to negative infinity)
+        # before the associated artist is drawn, and this method is meant
+        # to determine whether ``axes.get_tightbbox()`` may bypass drawing
+        ax = self.axes
+        clip_on = self.get_clip_on()
+        clip_box = self.get_clip_box()
+        clip_path = self.get_clip_path()
+        if ax is None or not clip_on or clip_box is None and clip_path is None:
+            return False
+        if clip_box is not None:
+            if not np.all(clip_box.extents == ax.bbox.extents):
+                return False
+        if clip_path is not None:
+            if not isinstance(clip_path, TransformedPatchPath):
+                return False
```

Comment:
I think it would be good if test coverage were 100% on this PR....

Reply:
I now include a test that `_fully_clipped_by_axes` returns `False` for non-default clipping situations.